src/stores/reactions.js

```diff
@@ -0,0 +1,318 @@
+			this.checkForExistence(token, messageId)
+
+			if (!this.reactions[token][messageId][reaction]) {
+				Vue.set(this.reactions[token][messageId], reaction, [])
+			}
+			const actors = this.reactions[token][messageId][reaction]
+			// Find if actor is already in the list
+			// This is needed when loading as revoking messages fully updates the list
+			const actorExists = actors.find(a => a.actorId === actor.actorId)
+			if (actorExists) {
+				return
+			}
+			actors.push(actor)
+			Vue.set(this.reactions[token][messageId], reaction, actors)
+		},
+
+		/**
+		 * Delete all reactions for a given message.
+		 *
+		 * @param {string} token The conversation token
+		 * @param {number} messageId The id of message
+		 *
+		 */
+		resetReactions(token, messageId) {
+			Vue.delete(this.reactions[token], messageId)
+		},
+
+		/**
+		 * Updates reactions for a given message.
+		 *
+		 * @param {object} payload action payload
+		 * @param {string} payload.token The conversation token
+		 * @param {number} payload.messageId The id of message
+		 * @param {object} payload.reactionsDetails The list of reactions with details for a given message
+		 *
+		 */
+		updateReactions({ token, messageId, reactionsDetails }) {
+			this.checkForExistence(token, messageId)
+
+			const storedReactions = this.reactions[token][messageId]
+
+			if (Object.keys(reactionsDetails).length === 0) {
+				this.resetReactions(token, messageId)
+				return
+			}
+
+			if (Object.keys(storedReactions).length === 0) {
+				Vue.set(this.reactions[token], messageId, reactionsDetails)
+				return
+			}
+
+			// Handle removed reactions
+			const removedReactions = Object.keys(storedReactions).filter(reaction => {
+				return !reactionsDetails[reaction]
+			})
+
+			removedReactions.forEach(reaction => {
+				this.removeReaction({ token, messageId, reaction })
+			})
+
+			// Add new reactions and/or update existing ones
+			Object.entries(reactionsDetails).forEach(([reaction, actors]) => {
+				if (!storedReactions[reaction] || JSON.stringify(actors) !== JSON.stringify(storedReactions[reaction])) {
+					this.addReaction({ token, messageId, reaction, actors })
+				}
+			})
+		},
+
+		/**
+		 * Process a reaction system message.
+		 *
+		 * @param {object} message the system message
+		 *
+		 */
+		processReaction(message) {
+			// 'reaction_deleted' is not handled because it is a message replacement
+			// for 'reaction' when the reaction is revoked, thus it doesn't exist anymore
+			if (message.systemMessage === 'reaction') {
+				const actorObject = {
+					actorDisplayName: message.actorDisplayName,
+					actorId: message.actorId,
+					actorType: message.actorType,
+					timestamp: message.timestamp,
+				}
+				this.addActorToReaction({
+					token: message.token,
+					messageId: message.parent.id,
+					reaction: message.message,
+					actor: actorObject,
+				})
+			} else if (message.systemMessage === 'reaction_revoked') {
```

Comment:
We can't really use that because it might not be accurate :
* Plain reactions can be `{ '🎄': 2, '🔥': 2, '🔒': 2 }` for example, which is a reflection of ` { '🎄': ['alice','bob'], '🔥': ['jake','bob'], '🔒': ['alice','bob'] }` ( the actual final plainReactions)
* Details reactions that are currently stored are `{ '🎄': ['alice','bob'], '🔥': ['alice','bob'], '🔒': ['alice','bob'] }`. 
* We see that the 'calculated' plain reactions of Detailed reactions are the same with the message plain reactions but in depth, they are different.
* Because of alice revoke reaction and at the same time, we have new reaction coming later from jake, alice won't be removed from the list and we will have` '🔥': ['jake','alice','bob']` along with count is 3 (ofc this will trigger fetch on 'after-show' which will adjust the count and the fetch to 2).

Reply:
Fair point, let's see how it feels and works